File: 1044.最大的质数/main.cpp
#include <iostream>
#include "cmath"

using namespace std;

int isprime(int num);

int main()
{
    int num,i;
    cin>>num;
    for(i=num-1;i>=2;i--)
    {
        if(isprime(i))
        {
            cout<<i<<endl;
            return 0;
        }
    }
    return 0;
}

int isprime(int num)
{
    int i;
    if(num==0||num==1)
        return 0;
    for(i=2;i<=sqrt(num);i++)
    {
        if(num%i==0)
            return 0;
    }
    return 1;
}

File: 1044.最大的质数/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        int N = new Scanner(System.in).nextInt();
        if (N <= 3) {
            System.out.println(1);
            return;
        }
        for (int i = N - 1; i > 1; i--) {
            if (isPrime(i)) {
                System.out.println(i);
                break;
            }
        }
    }

    public static boolean isPrime(int num) {
        if (num <= 1)
            return false;
        for (int i = 2; i <= ((num + 1) / 2); i++) {
            if (num % i == 0)
                return false;
        }
        return true;
    }
}
